selfdrive/test/test_valgrind_replay.py

```diff
@@ -0,0 +1,90 @@
+import os
+import threading
+import time
+import unittest
+import subprocess
+
+if "CI" in os.environ:
+  def tqdm(x):
+    return x
+else:
+  from tqdm import tqdm   # type: ignore
+
+import cereal.messaging as messaging
+from collections import namedtuple
+from tools.lib.logreader import LogReader
+from selfdrive.test.process_replay.test_processes import get_segment
+ProcessConfig = namedtuple('ProcessConfig', ['proc_name', 'pub_sub', 'ignore', 'command', 'path', 'segment', 'wait_for_response'])
+
+CONFIGS = [
+  ProcessConfig(
+    proc_name="ubloxd",
+    pub_sub={
+      "ubloxRaw": ["ubloxGnss", "gpsLocationExternal"],
+    },
+    ignore=[],
+    command="./ubloxd",
+    path="../locationd",
+    segment="0375fdf7b1ce594d|2019-06-13--08-32-25--3",
+    wait_for_response=True
+  ),
+]
+
+class TestValgrind(unittest.TestCase):
+
+  def extract_leak_sizes(self, log):
```

Comment:
this seems more complicated than it needs to be. can't we just check the valgrind exit code and use it's report if it fails?

Reply:
I get the sizes of leaks here, and also what type of leak they are. The file is intended to also be usable locally, not just in CI